File: examples/bench/native_string_slice.c
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(void) {
    const int64_t n = 100000;
    char **values = calloc((size_t)n, sizeof(*values));
    if (!values) abort();
    for (int64_t i = 0; i < n; ++i) {
        const char *source = (i % 2 == 0) ? "alpha" : "beta";
        values[i] = strdup(source);
        if (!values[i]) abort();
    }
    puts(values[0]);
    puts(values[n - 1]);
    printf("%lld\n", (long long)n);
    for (int64_t i = 0; i < n; ++i) free(values[i]);
    free(values);
    return 0;
}

File: examples/bench/native_string_slice.rs
fn main() {
    let n = 100_000usize;
    let mut values = Vec::with_capacity(n);
    for i in 0..n {
        values.push(if i % 2 == 0 { "alpha".to_owned() } else { "beta".to_owned() });
    }
    println!("{}", values[0]);
    println!("{}", values[n - 1]);
    println!("{}", values.len());
}
